### PR TITLE
Add the TSM identity to Darc

### PR DESCRIPTION
**Pull request by ineiti**
The Trusted Secure Module is used by DuoKey to sign messages using a threshold signature
scheme. This PR adds support for Darcs to verify those messages. It is used for the
Master Thesis protject of Pilar Marxer.

### CODE REVIEW

File: darc/darc.go [hunk 1]
```diff
@@ -964,6 +982,9 @@ func (id Identity) GetPublicBytes() []byte {
 		return buf
 	case 4:
 		return id.EvmContract.Address[:]
+	case 6:
+		buf := id.TSM.PublicKey
+		return buf
```
Comment (tharvik) [inline darc/darc.go:987]:
```suggestion
		return id.TSM.PublicKey
```

**Review by tharvik — APPROVED** [review]
some cosmetics changes, otherwise LGTM

File: darc/darc.go [hunk 2]
```diff
@@ -863,26 +875,18 @@ func (id Identity) Type() int {
...
-		return true
-	case id.X509EC != nil:
-		return true
-	case id.Proxy != nil:
-		return true
-	case id.EvmContract != nil:
-		return true
-	}
-	return false
+	return id.Type() != 0
```
Comment (cgrigis) [inline darc/darc.go:889]: Does this mean identities other than Darc are always Primary?

File: darc/darc.go [hunk 3]
```diff
@@ -937,6 +953,8 @@ func (id Identity) Verify(msg, sig []byte) error {
 		return id.Proxy.Verify(msg, sig)
 	case 4:
 		return id.EvmContract.Verify(msg, sig)
+	case 6:
+		return id.TSM.Verify(msg, sig)
```
Comment (cgrigis) [inline darc/darc.go:957]: Isn't `case 5:` missing?

File: darc/darc.go (hunk 1 above)
Comment (cgrigis) [inline darc/darc.go:987]: Isn't `case 5:` missing?

File: darc/darc.go [hunk 4]
```diff
@@ -1120,6 +1187,8 @@ func ParseIdentity(in string) (Identity, error) {
 		return parseIDProxy(fields[1])
 	case "evm_contract":
 		return parseIDEvmContract(fields[1])
+	case "tsm":
+		return parseIDTSM(fields[1])
 	default:
```
Comment (cgrigis) [inline darc/darc.go:1192]: Nothing for DID?

**Review by cgrigis — APPROVED** [review]
A few comments, I'll let you judge if they are relevant.

File: darc/darc.go (hunk 2 above)
Comment (ineiti) [reply darc/darc.go:889]: Yes - this is used to know which identities have counters. It should probably be renamed, but here I just thought it's useless to have this long `switch/case`, when a simple test would do...

File: darc/darc.go (hunk 3 above)
Comment (ineiti) [reply darc/darc.go:957]: The DID identity didn't get implemented completely, so it doesn't have a verification.

File: darc/darc.go (hunk 1 above)
Comment (ineiti) [reply darc/darc.go:987]: Yes, this is probably missing. I think the DID hasn't been completely implemented...

**Pull request merged**